asdf/asdf.py

```diff
@@ -90,7 +90,18 @@ def __init__(self, tree=None, uri=None, extensions=None, version=None,
 
         copy_arrays : bool, optional
             When `False`, when reading files, attempt to memmap underlying data
-            arrays when possible.
+            arrays when possible. Please note that the file must be opened
+            in "rw" mode; "r" will result a SIGSEGV.
```

Comment:
Not for this PR, but it seems like we should have better error handling for this case?

Reply:
Definitely. I had to debug segmentation faults otherwise.

Reply:
This is the current behavior:

```
asdf.open("/path", mode="r")  # SIGSEGV
asdf.open("/path", mode="rw")  # OK
with open("/path", mode="r") as file:
    asdf.open(file)  # SIGSEGV
with open("/path", mode="rw") as file:
    asdf.open(file)  # OK
```

So the comment is always effective as long as the option itself is effective (that is, no streaming and non-seekable files).

Reply:
Actually, I am surprised that this has not been reported yet. In my case I always use compressed blocks so never had that SIGSEGV before.

Reply:
Ah okay thanks. I've opened an issue to track this. We can leave the comment as-is here.

Reply:
So yeah, I should have added that the "rw" requirement is only for files with uncompressed arrays.

Reply:
Yes, I think I'm confused by this and will need to investigate further. Your example above suggests that opening any file without explicitly given a mode of `'rw'` leads to segfault, which doesn't make sense since this is most common use case, both in the test suite, and in our internal code base.

Reply:
Even with uncompressed arrays, that's the most common use case since it's the default.

Reply:
Sorry, but can we leave this comment out of this PR? I'm tracking it in #574 but I don't want to hold this PR up while I figure out what the issue is here.

Reply:
Sure